sql/firefox_desktop_visited_5_uri_mau28_by_dimensions_v1.sql

```diff
@@ -0,0 +1,52 @@
+WITH
+  _sample AS (
+  SELECT
+    submission_date,
+    -- most recent values for inclusion as output dimensions
+    days[SAFE_ORDINAL(1)] AS last_seen,
+    DATE_DIFF(submission_date, (
+      SELECT
+        -- bit-packed boolean array of days meeting the where condition
+        -- with most significant bit as the least recent day
+        SUM(1 << DATE_DIFF(submission_date, submission_date_s3, DAY))
+      FROM
+        UNNEST(days)
+      WHERE
+        -- results in NULL AS _days_visited_5_uri when condition not met by any item in days
+        scalar_parent_browser_engagement_total_uri_count_sum >= 5), DAY) AS _days_visited_5_uri
+  FROM
+    clients_monthly_v1
+  WHERE
+    submission_date = @submission_date )
+SELECT
+  submission_date,
+  last_seen.attribution.source,
+  last_seen.attribution.medium,
+  last_seen.attribution.campaign,
+  last_seen.attribution.content,
+  last_seen.country,
+  last_seen.distribution_id,
+  -- BIT_COUNT returns the number of set bits
+  -- 0x0FFFFFFF is a bit-mask for the lowest 28 bits
+  -- 0x7F is a bit-mask for the lowest 7 bits
+  -- Count the number of days in the window that a client visited 5 uri
+  BIT_COUNT(_days_visited_5_uri & 0x0FFFFFFF) AS visited_5_uri_frequency28,
+  BIT_COUNT(_days_visited_5_uri & 0x7F) AS visited_5_uri_frequency7,
```

Comment:
```suggestion
  SUM(BIT_COUNT(_days_visited_5_uri & 0x7F)) AS visited_5_uri_frequency7,
```

I think these need to be sums of bit counts. This column is representing the total number of client days meeting `visited_5_uri` for a particular grouping.

Reply:
nope, this should not be sum. it's a group-by column.

so if you visited_5_uri on 28 of the last 28 days, you would be counted in the group where this column is 28

Reply:
and then you could have a usage criterion for this column being greater than 1 for consistently active users, for instance

Reply:
Okay, wrapping my head around this, I think that means we could express Jesse's engagement metric as:

```
select
  submission_date,
  SUM(visited_5_uri_frequency7 * visited_5_uri_wau) / SUM(visited_5_uri_wau) AS visited_5_uri_engagement
FROM
  this_table
GROUP BY
  submission_date
```

Reply:
If Jesse's engagement metric is the average number of times in the last week an active user was active, then yes

Reply:
using the naming in #47 visited_5_uri_frequency7 would be days_visited_5_uri_last7